### PR TITLE
Don't show date of post when it is not set

### PR DESCRIPTION
**Pull request by fschrempf**
This moves duplicate code to a common partial and hides the date information when the date of a post is not set.

### CODE REVIEW

File: layouts/partials/author-date.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{{ with .Param "author" }}<span class="author" itemprop="author">{{ . }}</span>{{ end -}}
+{{ if and .Date (.Param "author") }} - {{ end -}}
+{{ with .Date }}<time class="created-date" datetime="{{ .Format "2006-01-02T15:04:05Z07:00" }}">{{ .Format ($.Param "dateformat" | default "2 January, 2006") }}</time>{{ end }}
```
Comment (frjo) [inline layouts/partials/author-date.html:3]: Can we change it back to "$.Param", see https://gohugo.io/functions/param/#prose

Otherwise nice improvement!

**Review by frjo — CHANGES_REQUESTED** [review]

File: layouts/partials/author-date.html (hunk 1 above)
Comment (fschrempf) [reply layouts/partials/author-date.html:3]: I didn't mean to change this. Thanks for spotting that! I pushed a fixup commit.

**Review by frjo — APPROVED** [review]

**Comment by frjo** [comment]
Thanks for helping making the Zen theme better!

**Comment by fschrempf** [comment]
> Thanks for helping making the Zen theme better!

You're welcome!

Just one nitpick/question: I was expecting that you would squash the fixup commit with the previous commit before merging. That would result in a cleaner history in master. Or do you rather want contributors to squash and force-push to their PR-branches when they do fixups?

**Comment by frjo** [comment]
No problems with force push to feature branches. Always needed when rebasing a feature branch. At work with many developers you do that a lot.

**Pull request merged**